base/exports.jl

```diff
@@ -70,6 +70,9 @@ export
     OrdinalRange,
     Pair,
     PartialQuickSort,
+    PerProcess,
+    PerTask,
+    PerThread,
```

Comment:
Bikeshedding in a thread..
Something seemed off about `Per*` on first impression. I think I was expecting there to be a `n` somewhere i.e. not specifically tied to _once_.

What about ProcessSingleton / ThreadSingleton / TaskSingleton ?

Reply:
Java used *Local naming convention for this construct, so in this context, that would be: ProcessLocal, ThreadLocal, TaskLocal

:-)

Reply:
In the condext of a `do` block, I think `OncePerProcess` is great.

"Once per process do function body" if a fairly accurate description of what this does.

Reply:
To help paint the bikeshed, I've come across a word that I think bears mentioning: "Hapax". It's probably a bit to esoteric to seriously consider, but it apparently refers to something (usually a word or expression) that occurs only once within a given context.

That seems like a pretty excellent semantic fit to me :grinning:

Other than that, `OncePerProcess`, `ProcessSingleton`, or `ProcessInitializer` seem like solid options to me.